### DISCUSSION TITLE
A simple script changing capitalized words of the small-case formatted text.

### DISCUSSION THREAD

**Discussion by ineuw** [autokey/autokey#923] [Show and tell]
Someone in this community was working with small-capitals type format and had the same issue as I. 

This script converts /incorrect/unwanted capitalization, when **small-capitals format** is applied to text. Since my previous contribution to the scripts was lost, I don't wish to post there and pasted it here.

My scripts use 'single quotes' to enclose for literal interpretation. Some symbols like **`< >`** in the text cause a script failure on Wikipedia pages.

```
# [ alt + super + c ]   lowercase connecting words on selected text formatted to small-case 
# 231121_1715 - alt-super-c-lcase.py - /home/ineuw/.config/autokey/data/english - Geany

copy_ = '<ctrl>+<insert>'
paste_ = '<shift>+<insert>'
ctrlhome_ = '<ctrl>+<home>'
time.sleep(0.05)

keyboard.send_keys(copy_)
time.sleep(0.05)

text_ = clipboard.get_selection()
time.sleep(0.05)

text_ = text_.replace(' A ', ' a ')
time.sleep(0.05)

text_ = text_.replace(' An ', ' an ')
time.sleep(0.05)

text_ = text_.replace(' And ', ' and ')
time.sleep(0.05)

text_ = text_.replace(' As ', ' as ')
time.sleep(0.05)

text_ = text_.replace(' For ', ' for ')
time.sleep(0.05)

text_ = text_.replace(' From ', ' from ')
time.sleep(0.05)

text_ = text_.replace(' In ', ' in ')
time.sleep(0.05)

text_ = text_.replace(' Of ', ' of ')
time.sleep(0.05)

text_ = text_.replace(' The ', ' the ')
time.sleep(0.05)

text_ = text_.replace(' To ', ' to ')
time.sleep(0.05)

text_ = text_.replace(' Vs ', ' vs ')
time.sleep(0.05)

text_ = text_.replace(' With ', ' with ')
time.sleep(0.05)

# place result to the clipboard
clipboard.fill_clipboard(text_)
time.sleep(0.05)

# paste clipboard contents
keyboard.send_keys(paste_)
time.sleep(0.05)

# move focus to top left corner
keyboard.send_keys(ctrlhome_)

```

**Comment by josephj11** [comment]
I have no idea what "small-capitals" is.

If your script was lost from our wiki, I'm sure it was an accident. We moved a lot of things around to organize it better and may have missed yours.

Your code looks like a perfect use case for a Python dictionary with a loop through all its entries.

Why did you add delays after each replacement? I'm pretty sure those statements are synchronous (don't return until they're done).

**Reply by ineuw** [reply]
@josephj11, Small-caps are uppercase characters in lowercase dimension (height). [small caps](https://en.wikipedia.org/wiki/Small_caps) and it's used for emphasis. 

A text-formatting template that converts to "small-caps" also CAPITALIZES the first character of each word and this necessitates a script to "lower the case" of connecting words.

This is the reason for my script. The chapter II title "Review of the Situation" [on this page](https://en.wikisource.org/wiki/History_of_Woman_Suffrage/Volume_4#Contents) is an example where the words "of the" were changed to lower case according to rules.

[additional explanation](https://en.wikisource.org/wiki/User:Ineuw/notes3)

I decided against a dictionary because the number of words.

Timing of interpreted code is truly dependent on processor speed and memory. I use a very fast processor (4Ghz Intel processor + 16GB RAM ). Python being an interpreted language, the delay ensures that the execution is **complete** at all times. I arrived at .050 seconds by stepped incrementation of .010 after it kept failing.

**Reply by josephj11** [reply]
If you ever find an explanation of why that delay is needed, I'd like to see it. It doesn't seem intuitive. But, especially with AutoKey, empirical results rule.
